splunk/update-demos.sh

```diff
@@ -0,0 +1,108 @@
+    yq eval -i '.services.otelcol.ports += [ "6060" ]' "$SPLUNK_DOCKER_COMPOSE_PATH"
+    yq eval -i '.services.otelcol.ports += [ "9080" ]' "$SPLUNK_DOCKER_COMPOSE_PATH"
+    yq eval -i '.services.otelcol.ports += [ "9411" ]' "$SPLUNK_DOCKER_COMPOSE_PATH"
+    yq eval -i '.services.otelcol.ports += [ "9943" ]' "$SPLUNK_DOCKER_COMPOSE_PATH"
+
+    echo "Completed updating docker-compose.yml for the OpenTelemetry demo app!"
+}
+
+function update_otel_demo_k8s {
+    K8S_PATH=${K8S_PATH:-"$SCRIPT_DIR/../kubernetes/opentelemetry-demo.yaml"}
+    SPLUNK_K8S_PATH=${SPLUNK_K8S_PATH:-"$SCRIPT_DIR/../splunk/opentelemetry-demo.yaml"}
+
+    # delete any older versions of the Splunk K8s file
+    [ -e "$SPLUNK_K8S_PATH" ] && rm "$SPLUNK_K8S_PATH"
+
+    # Download the YAML file
+    curl -L https://raw.githubusercontent.com/open-telemetry/opentelemetry-demo/main/kubernetes/opentelemetry-demo.yaml \
+        > "$K8S_PATH"
+
+    cp "$K8S_PATH" "$SPLUNK_K8S_PATH"
+
+    # delete the opentelemetry-demo-otelcol ServiceAccount, ConfigMap, Service, and Deployment objects
+    yq eval -i 'select(.kind != "ServiceAccount" or .metadata.name != "opentelemetry-demo-otelcol")' "$SPLUNK_K8S_PATH"
+    yq eval -i 'select(.kind != "ConfigMap" or .metadata.name != "opentelemetry-demo-otelcol")' "$SPLUNK_K8S_PATH"
+    yq eval -i 'select(.kind != "Service" or .metadata.name != "opentelemetry-demo-otelcol")' "$SPLUNK_K8S_PATH"
+    yq eval -i 'select(.kind != "Deployment" or .metadata.name != "opentelemetry-demo-otelcol")' "$SPLUNK_K8S_PATH"
+
+    # replace the OTEL_COLLECTOR_NAME environment variable
+    # with a NODE_IP environment variable for all containers
+    #      - name: NODE_IP
+    #        valueFrom:
+    #          fieldRef:
+    #            fieldPath: status.hostIP
+
+    # start by replacing the environment variable name
+    yq eval -i ' (.. | select(tag == "!!str")) |= sub("OTEL_COLLECTOR_NAME", "NODE_IP")' "$SPLUNK_K8S_PATH"
+
+    # then use sed to replace the value
+    # (yq was not used due to an issue that added extraneous elements to the file)
+    SEARCH_VAL="value: 'opentelemetry-demo-otelcol'"
+    REPLACE_VAL='valueFrom: \
+                fieldRef: \
+                  fieldPath: status.hostIP'
+
+    sed -i '' "s/${SEARCH_VAL}/${REPLACE_VAL}/g" "$SPLUNK_K8S_PATH"
+
+    # append the deployment.environment resource attribute
+    # - name: OTEL_RESOURCE_ATTRIBUTES
+    #   value: service.name=$(OTEL_SERVICE_NAME),service.namespace=opentelemetry-demo,deployment.environment=development
+    yq eval -i ' (.. | select(tag == "!!str")) |= sub("service.namespace=opentelemetry-demo", "service.namespace=opentelemetry-demo,deployment.environment=development")'  "$SPLUNK_K8S_PATH"
+
+    echo "Completed updating the kubernetes/opentelemetry-demo.yaml for the OpenTelemetry demo app!"
+}
+
+# ---- OpenTelemetry Demo Update ----
+update_otel_demo_docker
+update_otel_demo_k8s
```

Comment:
We should add one more function in this file. We should add a update_root_readme function that appends to or updates the root README.md.
- The root readme should have an obvious reference to the splunk/README.md
- We want users to know our Splunk resources exist in a certain location.

I can help out with this in a follow up PR if you don't get to it. This PR is already pretty big.

Reply:
Great idea, thanks.  I've applied this change.